### PR TITLE
[Numpy] Support `mxnet.numpy.ndarray.diagonal()`

### PR DESCRIPTION
**Pull request by shenfei**
## Description ##
Support `mxnet.numpy.ndarray.diagonal()` and fix https://github.com/apache/incubator-mxnet/issues/18301

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- `mxnet.numpy.ndarray.diagonal()`
- `unittest/test_numpy_op.py::test_np_diagonal`

## Comments ##
This is my first PR for MXNet, so I’m not sure if my unit test’s change is correct.

### CODE REVIEW

File: tests/python/unittest/test_numpy_op.py [hunk 1]
```diff
@@ -9244,21 +9243,24 @@ def hybrid_forward(self, F, a):
...
     rtol = 1e-2 if dtype == np.float16 else 1e-3
     atol = 1e-4 if dtype == np.float16 else 1e-5
-    shape = config[0]
-    axis = config[1]
-    axis1 = axis[0]
-    axis2 = axis[1]
+    shape, (axis1, axis2) = config
     x = np.random.uniform(-5.0, 5.0, size=shape).astype(dtype)
     x.attach_grad()
     test_diagonal = TestDiagonal(k, axis1, axis2)
```
Comment (xidulu) [inline tests/python/unittest/test_numpy_op.py:9249]: The test case you modified did not cover the hybridize case (i.e. the symbol mode), so I suggest changing the function call at https://github.com/apache/incubator-mxnet/blob/master/tests/python/unittest/test_numpy_op.py#L9243 as well to test if your patch is fully compatible with hybrid block.

Comment (shenfei) [reply tests/python/unittest/test_numpy_op.py]: @xidulu Thanks for the review. I’ve added two new commits https://github.com/apache/incubator-mxnet/pull/19285/commits/c9c8aa7df44c7936c4e54c1716ac1d5cac6c2f4a and https://github.com/apache/incubator-mxnet/pull/19285/commits/8ae59cefad67fa1eea1cc5a6bc0033dc6faa570a to cover the hybridize case.

**Comment by xidulu** [comment]
@mxnet-bot run ci [windows-gpu]

**Comment by shenfei** [comment]
@mxnet-bot run ci [windows-gpu]

**Comment by xidulu** [comment]
@mxnet-bot run ci [edge]

**Review by xidulu — APPROVED** [review]
LGTM
Thanks for ur contribution.

**Pull request merged**